process/transaction/transactionCostEstimator.go

```diff
@@ -193,9 +194,10 @@ func (tce *transactionCostEstimator) getTxGasLimit(txSender []byte) (gasLimit ui
 		return
 	}
 
-	maxGasLimitPerBlockBig := big.NewInt(0).SetUint64(maxGasLimitPerBlock)
 	accountSenderBalance := accountSender.GetBalance()
-	if accountSenderBalance.Cmp(maxGasLimitPerBlockBig) < 0 && accountSenderBalance.Uint64() != 0 {
+	tx.GasLimit = maxGasLimitPerBlock
+	txFee := tce.feeHandler.ComputeTxFee(tx)
+	if accountSenderBalance.Cmp(txFee) < 0 && accountSenderBalance.Uint64() != 0 {
```

Comment:
compoare on big ints as this might overflow

Reply:
done.